### ISSUE TITLE
add option to reset reader to first row

### ISSUE THREAD

**Issue by LouisFoucard** [uber/petastorm#343]
In some scenario, it could be useful to have an option to reset the reader to the first row.

Example: 
using the petastorm reader for evaluation during training:
Let's say that we have a big validation set, and we only want to use the first few 100 rows to compute some metrics every 10k training step (and we don't want to create a subset dataset specifically for this). For the metric to be meaningful, we would need to operate on the same data each time.
If we have a reset option, we can run for the 1st 100 rows, generate the metric, then reset the reader for the next evaluation run.
